src/api-client/files-api.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { ApiClient } from './api-client';
+import { FilesApi } from './files-api';
+
+const apiConfig = { BATCH_DUMP_TIMEOUT_MS: 1, BATCH_URL: '/file-batch-api' };
+jest.mock('./consts', () => apiConfig);
+
+jest.mock('./api-client', () => ({
+  ApiClient: {
+    get: jest.fn()
+  }
+}));
+
+const defaultFiles = ['fileid1', 'fileid2', 'fileid3'];
+
+describe('Files API', () => {
```

Comment:
This test suite does have quite a few tests. I think I tested everything that came to my head.

Reply:
I also tested quite a few of corner case and error scenarios.